### ISSUE TITLE
Resolving Func use wrong constructor

### ISSUE THREAD

**Issue by mgth** [z4kn4fein/stashbox#102]
resolving Func with arguments on multi constructors class will sometimes select wrong constructor.

```csharp
    class ClassA
    { }

    class ClassB
    {
    }

    class InjectedClass
    {
        public ClassA ClassA;
        public ClassB ClassB;

        public InjectedClass(ClassA classA)
        {
            ClassA = classA;
        }

        public InjectedClass(ClassB classB)
        {
            ClassB = classB;
        }
    }

    public class ConstructorSelection
    {
        [Fact]
        public void Test()
        {
            var container = new StashboxContainer();
            container.RegisterAssemblyContaining<ClassA>();

            var funcA = container.Resolve<Func<ClassA, InjectedClass>>();
            var classA = container.Resolve<ClassA>();
            var injectedA = funcA(classA);

            var funcB = container.Resolve<Func<ClassB, InjectedClass>>();
            var classB = container.Resolve<ClassB>();
            var injectedB = funcB(classB);

            Assert.Equal(classA,injectedA.ClassA);
            Assert.Equal(classB,injectedB.ClassB); //first constructor used, with newly created ClassA and classB not used
        }
    }
```

**Comment by z4kn4fein** [comment]
Hi there, thanks for reporting! I pushed a fix for this issue, it'll be available as a pre-release version within a few minutes. Could you please run a test with it? Thanks!

**Comment by mgth** [comment]
Thanks a lot it works great (but i have another one)!

**Comment by z4kn4fein** [comment]
Released in [v3.6.3](https://github.com/z4kn4fein/stashbox/releases/tag/3.6.3)
